libbeat/mapping/field.go

```diff
@@ -110,6 +110,9 @@ func (d *DynamicType) Unpack(s string) error {
 
 // Validate ensures objectTypeParams are not mixed with top level objectType configuration
 func (f *Field) Validate() error {
+	if err := f.validateType(); err != nil {
+		return err
```

Comment:
What about returning here `error trying to check 'type' field in 'fields.yml' fike '%s'. Check the 'fields.yml' in the module you are modifying for an incorrect or unknonwn 'type' value`

Reply:
Ok, I have reordered error wrapping a little bit, this is what is printed now:
```
Generated global fields.yml file for metricbeat is invalid: incorrect type configuration for field 'max.bytes': unexpected formatter for number type, expected one of: string, url, bytes, duration, number, percent, color accessing '52.fields.0.fields.0.fields.1.fields.2.fields.0' (source:'fields.yml')
exit status 3
```
This is not ideal, but I haven't found a way to get more info from the underlying parsing library. In any case it is much better than what we have now.

Reply:
Totally agree. It's much better. Sorry for asking you a bit more :sweat_smile: maybe it's worth to just add `(source: 'beats/metricbeat/module/{your-module}/_meta/fields.yml')` or something similar. A new contributor won't know where the fields.yml is gonna be located.

Reply:
This part of the message is generated by the library, we are a bit limitated by that.

Also take into account that code in this PR only checks the global `fields.yml`, `fields.yml` files in modules and metricsets are not valid YAML by design, so they are not so easy to parse and check (and we don't). They are concatenated to create the global `fields.yml` and only this should be a valid YAML, but we are actually not even checking that, and this is not detected at all unless something mysteriously breaks in some integration test.

In summary, yes, at the moment it is not possible to easily know the module of the fields causing the problems, but hopefully a PR will only touch a limited number of fields, so it shouldn't be so hard to spot.